### PR TITLE
Add support for compute/DiskEncryptionSets

### LINKED ISSUE #2927: Feature: DiskEncryptionSets Support
<Describe the feature request>

Hi,

We would like to provision DiskEncryptionSets through ASO. It's not managable when running through az or arm templates right now for us.

### PR DESCRIPTION
**Pull request by super-harsh**
Closes #2927 

**What this PR does / why we need it**:

This PR adds support for [Microsoft.Compute diskEncryptionSets](https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/diskencryptionsets?pivots=deployment-language-bicep)

**Special notes for your reviewer**:

Hacking around KeyVault/Keys with SDK in controller test for now until we support #3188 

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: v2/api/compute/v1api20220702/disk_encryption_set_types_gen.go [hunk 1]
```diff
@@ -0,0 +1,2481 @@
...
+	// AzureName: The name of the resource in Azure. This is often the same as the name of the resource in Kubernetes but it
+	// doesn't have to be.
+	AzureName string `json:"azureName,omitempty"`
+
+	// EncryptionType: The type of key used to encrypt the data of the disk.
+	EncryptionType *DiskEncryptionSetType `json:"encryptionType,omitempty"`
+
+	// FederatedClientId: Multi-tenant application client id to access key vault in a different tenant. Setting the value to
+	// 'None' will clear the property.
+	FederatedClientId *string `json:"federatedClientId,omitempty"`
```
Comment (theunrepentantgeek) [inline v2/api/compute/v1api20220702/disk_encryption_set_types_gen.go:344]: Should this optionally support coming from a configmap?

**Review by theunrepentantgeek — COMMENTED** [review]
I'm concerned by the number of modified deepcopy files, as most are unrelated to your change. I've tagged one property that should probably be configurable, but other than that I'm generally happy. Will approve once we've sorted the deepcopy problem.

File: v2/api/compute/v1api20220702/disk_encryption_set_types_gen.go (hunk 1 above)
Comment (super-harsh) [reply v2/api/compute/v1api20220702/disk_encryption_set_types_gen.go]: Thanks for spotting. Done!

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/internal/controllers/crd_compute_diskencryptionset_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+	// Ensure delete
+	exists, retryAfter, err := tc.AzureClient.HeadByID(tc.Ctx, armId, string(compute.APIVersion_Value))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(retryAfter).To(BeZero())
+	tc.Expect(exists).To(BeFalse())
+}
+
+func newVaultForDiskEncryptionSet(name string, tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup) *keyvault.Vault {
+
```
Comment (matthchr) [inline v2/internal/controllers/crd_compute_diskencryptionset_test.go:122]: minor: unneeded newline?

**Review by matthchr — APPROVED** [review]

**Pull request merged**